llmebench/models/HuggingFace.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import json
+import requests
+from llmebench.models.model_base import ModelBase
+import os
+
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_random_exponential,
+)
+
+API_TOKEN = os.environ["HUGGINGFACE_API_TOKEN"]
+
+
+def log_retry(retry_state):
+    if retry_state.attempt_number == 1:
+        return
+    logging.warning(
+        f"Request failed, retry attempt {retry_state.attempt_number}...")
+
+
+class HuggingFace(ModelBase):
+    def __init__(self, inference_api_url, max_tries=5, retry_exceptions=()):
```

Comment:
We should not really be taking in retry_exceptions from the user ideally, this is an internal implementation detail; Also max_tries can just be passed along to the `ModelBase` constructor by using **kwargs (See `BLOOMPetal.py:25`)